lib/userAgent.js

```diff
@@ -34,6 +36,11 @@ module.exports = function (userAgentString) {
   this.uaOSVersion = getVersion(userAgentData.os) || null
   this.uaDeviceType = getDeviceType(userAgentData) || null
 
+  if (! this.uaBrowser && ! this.uaOS) {
+    // In the worst case, fall back to a truncated user agent string
+    this.uaBrowser = truncate(userAgentString || '', log)
```

Comment:
> perhaps the truncation could happen in the client

I did consider that but decided it against it for the following reasons, which may or may not be good ones:

* There's already an arbitrary limit set in the db anyway because `uaBrowser` is `VARCHAR(255)`.

* Structurally, this felt like the correct place to do it because its our one-and-only point (I think?) at which we parse the user agent string and break it down into the constituent parts. After this, we only handle the parsed fields (or the fallback).

* Assuming the front-end would truncate using `overflow: hidden; text-overflow: ellipsis;` in CSS, the code here can be a little bit smarter about truncating the string at a sensible place (i.e. when there is a closing parenthesis, if there is one).

None of them are massively compelling though, I'm happy to change it if you think this is the wrong place.